File: console_edition_Java/MyChess/TransTable.java
package MyChess;

import MyChess.TableCell;

/**
 * TransTable
 */
public class TransTable {
    private static final int TABLE_SIZE = 1 << 20;// 1,048,576
    public static TableCell[] cells;

    static {
        cells = new TableCell[TABLE_SIZE];
        for (int i = 0; i < TABLE_SIZE; i++) {
            cells[i] = new TableCell();
        }
        System.out.println("ok");
    }

    public static boolean isExist(long hashCode) {
        int index = (int) (((hashCode % TABLE_SIZE) + TABLE_SIZE) % TABLE_SIZE);
        return cells[index].isValid;
    }

    public static TableCell get(long hashCode) {
        int index = (int) (((hashCode % TABLE_SIZE) + TABLE_SIZE) % TABLE_SIZE);
        return cells[index];
    }
}

File: src/GameAlgo/TransTable.js
//import TableCell from "./TableCell";

const TABLE_SIZE = 1 << 21;
let cells = new Array(TABLE_SIZE);
cells.fill(null, 0, TABLE_SIZE);
//console.log("transtable size:" + TABLE_SIZE);

/**
 * 获取对应的置换表表项
 * @param {Number} hashCode 棋盘哈希值
 * @returns {TableCell} 置换表表项
 */
function getCell(hashCode) {
    let index = ((hashCode % TABLE_SIZE) + TABLE_SIZE) % TABLE_SIZE;
    return cells[index];
}

/**
 * 新建对应的置换表表项
 * @param {Number} hashCode 棋盘哈希值
 * @param {TableCell} cell 新建的置换表表项
 */
function setCell(hashCode, cell) {
    let index = ((hashCode % TABLE_SIZE) + TABLE_SIZE) % TABLE_SIZE;
    cells[index] = cell;
}

export { getCell, setCell };